src/Router.sol

```diff
@@ -456,10 +533,102 @@ contract Router is Ownable, ReentrancyGuard, IRouter {
     /// @notice Withdraws verification fees to a specified address
     /// @param token The token address of the withdrawn fees
     /// @param to The address receiving the withdrawn fees
-    function withdrawVerificationFee(address token, address to) external onlyOwner nonReentrant {
-        uint256 amountOut = totalVerificationFeeBalance[token];
+    function withdrawVerificationFee(address token, address to) external onlyAdmin nonReentrant {
+        uint256 amount = totalVerificationFeeBalance[token];
+        require(amount > 0, ErrorsLib.ZeroAmount());
         totalVerificationFeeBalance[token] = 0;
-        IERC20(token).safeTransfer(to, amountOut);
-        emit VerificationFeeWithdrawn(token, to, amountOut);
+        IERC20(token).safeTransfer(to, amount);
+        emit VerificationFeeWithdrawn(token, to, amount);
+    }
+
+    // ---------------------- Scheduled Upgrade Functions ----------------------
+
+    /// @notice Schedules a contract upgrade
+    /// @param _newImplementation The address of the new implementation contract
+    /// @param _upgradeCalldata The calldata to be sent to the new implementation
+    /// @param _upgradeTime The time at which the upgrade can be executed
+    function scheduleUpgrade(address _newImplementation, bytes calldata _upgradeCalldata, uint256 _upgradeTime)
+        external
+        onlyAdmin
```

Comment:
thie defeats the purpose of the BLS sig; imo we should allow a valid BLS sig alone to upgrade

Reply:
That's right. The `onlyAdmin` modifier has been removed and replaced with BLS signature verification for the `"schedule"` action.